crates/ruff/src/checkers/ast/mod.rs

```diff
@@ -2084,8 +2084,11 @@ where
             }) => {
                 self.visit_boolean_test(test);
 
-                if flake8_type_checking::helpers::is_type_checking_block(&self.semantic_model, test)
-                {
+                if analyze::typing::is_type_checking_block(&self.semantic_model, test) {
```

Comment:
Should we pass the entire if statement here? it would prevent from calling this function with e.g. a `test` condition from a while or any other expression (where the result would not make sense)